core/src/main/java/org/aerogear/mobile/core/metrics/MetricsPublisher.java

```diff
@@ -1,12 +1,18 @@
 package org.aerogear.mobile.core.metrics;
 
+import android.support.annotation.NonNull;
+import android.support.annotation.Nullable;
+
+import org.aerogear.mobile.core.Callback;
+
 public interface MetricsPublisher {
 
     /**
      * Allows to publish metrics to external source
-     * 
+     *
      * @param metrics a array of metrics objects to publish
+     * @param callback callback of the publication
      */
-    void publish(Metrics... metrics);
+    void publish(@NonNull final Metrics[] metrics, @Nullable final Callback callback);
```

Comment:
@ziccardi I have added `publish(Metrics... metrics);` to make easy to do `publish(new MyAlsoMetrics(), new MyNotSoAwesomeMetrics())`. Since it's not going to be used anymore, should we replace `Metrics[]` by a `List<Metrics>`?

Reply:
I say we merge this once for all and after that we think about changing the array for collection.

Reply:
np for me!